### PR TITLE
Allow global drawables to be destroyed when there is no list + windows crash fixes

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #2073
Fix: #2061 
Fix: #2045 

@elsemieni can you confirm this fixes the crash you found? I can't reproduce it so while I suspect this is the cause I can't prove it.

### CODE REVIEW

**Comment by elsemieni** [comment]
@fmatthew5876 Used associated jenkins PR build (Win32). After splash logo it throws me an assertion error window (stdout/stderr doesn't show me any interesting about it).

```
Program: C:\msys32\home\elsemieni\nether\SCH\Player.exe
File: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\xstring
Line: 1792

Expression: cannot dereference string iterator because it is out of range (e.g. an end iterator)
```

**Comment by fmatthew5876** [comment]
Ok I added another fix which hopefully covers that crash.

**Comment by elsemieni** [comment]
Same assertion failure :( 
This time I attached a debugger and well... I wonder there's a bizarre stack trace over there 
image

**Comment by fmatthew5876** [comment]
Alright I went ahead and fixed all the windows crashes. Tested it myself on windows and seems to work now

**Comment by Ghabry** [comment]
This still fails with "error C2015: too many characters in constant".

I believe the problem is that "/utf-8" (interpret literals as utf-8) is not passed to the unit tests.

Untested (not at my laptop right now) solution:
``target_compile_options(${PROJECT_NAME} PRIVATE "/utf-8")``
but it should be ``PUBLIC`` otherwise this compile flag is not propagated to others that depend on ``${PROJECT_NAME}``.

**Comment by Ghabry** [comment]
Jenkins: Test this please <3

**Comment by Ghabry** [comment]
This patch fixes the Windows build (likely needs Workspace wipe):

``patch -Np1 < patch.diff``

```patch.diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77051b94..1c99be62 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -664,11 +664,6 @@ if(${PLAYER_AUDIO_BACKEND} MATCHES "^(SDL.*|libretro)$")
 	endif()
 endif()
 
-# MSVC: Set UTF-8 encoding for all files (important for rtp table)
-if(MSVC)
-	target_compile_options(${PROJECT_NAME} PRIVATE "/utf-8")
-endif()
-
 # Executable
 if (${PLAYER_BUILD_EXECUTABLE})
 	add_executable(${PROJECT_NAME}_exe WIN32 "src/main.cpp")
diff --git a/builds/cmake/Modules/ConfigureWindows.cmake b/builds/cmake/Modules/ConfigureWindows.cmake
index b4f21bb2..2cebb8cb 100644
--- a/builds/cmake/Modules/ConfigureWindows.cmake
+++ b/builds/cmake/Modules/ConfigureWindows.cmake
@@ -43,6 +43,8 @@ if(MSVC)
 
 	option(MSVC_MULTICORE "MSVC: Build using multiple cores (/MP)" ON)
 	if (MSVC_MULTICORE)
-		add_compile_options(/MP)
+		add_compile_options("/MP")
 	endif()
+
+	add_compile_options("/utf-8")
 endif()
```

**Comment by fmatthew5876** [comment]
Added your patch

**Comment by Ghabry** [comment]
Now it builds but there are still end-derefs.
I attached a debugger: You missed two occurences of ``text.end()`` in utf.cpp

**Comment by fmatthew5876** [comment]
jenkins: test this please

**Comment by Ghabry** [comment]
Jenkins: Test this please

**Comment by Ghabry** [comment]
Jenkins: Test this please

File: builds/cmake/Modules/ConfigureWindows.cmake [hunk 1]
```diff
@@ -43,6 +43,8 @@ if(MSVC)
 
 	option(MSVC_MULTICORE "MSVC: Build using multiple cores (/MP)" ON)
 	if (MSVC_MULTICORE)
-		add_compile_options(/MP)
+		add_compile_options("/MP")
 	endif()
+
+	add_compile_options("/utf-8")
```
Comment (carstene1ns) [inline builds/cmake/Modules/ConfigureWindows.cmake]: What I dislike here, is that the comment is lost.

Comment (Ghabry) [reply builds/cmake/Modules/ConfigureWindows.cmake]: Valid concern, maybe make the comment more generic now "Interpret character literals as UTF-8"

Comment (fmatthew5876) [reply builds/cmake/Modules/ConfigureWindows.cmake]: Added comment in #2017

**Pull request merged**